### PR TITLE
fix(cors): compatible with scenarios where origin is modified

### PR DESCRIPTION
**Pull request by zuiyangqingzhou**
In some scenarios, if you modify the origin using proxy_rewrite plugin, the cors will not be handled correctly

### Pre-submission checklist:

* [ ] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: apisix/plugins/cors.lua [hunk 1]
```diff
@@ -227,14 +227,15 @@ end
 
 
 function _M.rewrite(conf, ctx)
+    ctx["raw-request-origin"] = core.request.header(ctx, "Origin")
```
Comment (spacewander) [inline apisix/plugins/cors.lua:230]:
```suggestion
    ctx.original_request_origin = core.request.header(ctx, "Origin")
```
would be better.

Please also add a comment about this change.

**Review by tokers — COMMENTED** [review]
It seems that all other CORS-related headers also suffer from this change.

File: apisix/plugins/cors.lua (hunk 1 above)
Comment (zuiyangqingzhou) [reply apisix/plugins/cors.lua:230]: OK

**Comment by zuiyangqingzhou** [comment]
> 



> It seems that all other CORS-related headers also suffer from this change.

Sorry, I don't quite understand what you mean, can you list some cases?

**Comment by tokers** [comment]
> 

Headers like `Access-Control-Expose-Methods`, `Access-Control-Max-Age` will also be modified by `proxy-rewrite` plugin,

**Comment by zuiyangqingzhou** [comment]
> > 
> 
> Headers like `Access-Control-Expose-Methods`, `Access-Control-Max-Age` will also be modified by `proxy-rewrite` plugin,

But `proxy-rewrite` plugin works in the rewrite phase, `Access-Control-Expose-Methods`, `Access-Control-Max-Age` is written to the response header in the header_filter phase,  I understand that it should not be affected

**Review by spacewander — APPROVED** [review]

File: t/plugin/cors.t [hunk 2]
```diff
@@ -927,3 +927,70 @@ Access-Control-Max-Age:
...
+                            "allow_origins": "http://sub.domain.com",
+                            "allow_methods": "GET,POST",
+                            "allow_headers": "headr1,headr2",
+                            "expose_headers": "ex-headr1,ex-headr2",
+                            "max_age": 50,
+                            "allow_credential": true
+                        },
+                        "proxy-rewrite": {
+                            "Origin": "http://example.com"
+                        }
```
Comment (tzssangglass) [inline t/plugin/cors.t:952]:
```suggestion
                        "proxy-rewrite": {
                            "headers": {
                                 "Origin": "http://example.com"
                            }
                        }
```

Modifying the header in proxy-rewrite should be configured like this?

Comment (zuiyangqingzhou) [reply t/plugin/cors.t:952]: oh,  I will fix it

**Comment by tokers** [comment]
> > > 
> > 
> > 
> > Headers like `Access-Control-Expose-Methods`, `Access-Control-Max-Age` will also be modified by `proxy-rewrite` plugin,
> 
> But `proxy-rewrite` plugin works in the rewrite phase, `Access-Control-Expose-Methods`, `Access-Control-Max-Age` is written to the response header in the header_filter phase, I understand that it should not be affected

Oh, forget it, it's my bad.

**Pull request merged**